### PR TITLE
restrict `postNewContentFragment` to `ContentFragmentInputWithFileIdType`

### LINKED ISSUE #9171: Ensure all attachments are created with files

### PR DESCRIPTION
**Pull request by spolu**
## Description

Fixes: https://github.com/dust-tt/dust/issues/9171

Introduce `toFileContentFragment` that transforms a content-based contentFragment into a file-based one. Use it everywhere we can.

## Risk

Low, tested locally

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/lib/api/assistant/email_trigger.ts [hunk 1]
```diff
@@ -309,46 +309,39 @@ export async function triggerFromEmail({
...
-    const file = await FileResource.makeNew({
-      contentType: "text/plain",
+    const cfRes = await toFileContentFragment(auth, {
+      contentFragment: {
+        title: `Email thread: ${email.subject}`,
+        content: restOfThread,
+        contentType: "text/plain",
+        url: null,
+      },
       fileName: `email-${email.subject}.txt`,
```
Comment (Fraggle) [inline front/lib/api/assistant/email_trigger.ts:319]: nit: the filename might be truncated to 256 characters.

Comment (spolu) [reply front/lib/api/assistant/email_trigger.ts:319]: will call it email-thread.txt

File: types/src/front/files.ts [hunk 2]
```diff
@@ -191,6 +191,20 @@ export function isSupportedDelimitedTextContentType(
   );
 }
 
+export function extensionsForContentType(
+  contentType: SupportedFileContentType
+): string[] {
+  if (isSupportedPlainTextContentType(contentType)) {
+    return [...supportedPlainText[contentType]];
```
Comment (Fraggle) [inline types/src/front/files.ts:198]: why not `return supportedPlainText[contentType];` ?

Comment (spolu) [reply types/src/front/files.ts:198]: they are const

**Review by Fraggle — APPROVED** [review]
LGTM, thanks for taking of aligning this.

**Pull request merged**